client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts

```diff
@@ -84,15 +84,20 @@ export default function getThankYouPageUrl( {
 	debug( 'starting getThankYouPageUrl' );
 
 	// If we're given an explicit `redirectTo` query arg, make sure it's either internal
-	// (i.e. on WordPress.com), or a Jetpack or WP.com site's block editor (in wp-admin).
-	// This is required for Jetpack's (and WP.com's) paid blocks Upgrade Nudge.
-	if ( redirectTo && ! isExternal( redirectTo ) ) {
-		debug( 'has external redirectTo, so returning that', redirectTo );
-		return redirectTo;
-	}
+	// (i.e. on WordPress.com), the same site as the cart's site, or a Jetpack or WP.com
+	// site's block editor (in wp-admin). This is required for Jetpack's (and WP.com's)
+	// paid blocks Upgrade Nudge.
 	if ( redirectTo ) {
 		const { protocol, hostname, port, pathname, query } = parseUrl( redirectTo, true, true );
 
+		if ( resemblesUrl( redirectTo ) && hostname && urlToSlug( hostname ) === siteSlug ) {
+			debug( 'has same site redirectTo, so returning that', redirectTo );
+			return redirectTo;
+		}
+		if ( ! isExternal( redirectTo ) ) {
+			debug( 'has external redirectTo, so returning that', redirectTo );
```

Comment:
My debug line here was wrongly worded (despite the fact that the comment above is correct); this condition actually means it's _not_ external. My fault but if you could correct my mistake it'd be appreciated.